edit/src/lib.rs

```
@@ -0,0 +1,108 @@
+    the current project is edited.
+
+Options
+
+    --help              Print help
+"#,
+};
+
+#[derive(Default, Debug, Eq, PartialEq)]
+pub struct Options {
+    pub urn: Option<Urn>,
+}
+
+impl Args for Options {
+    fn from_args(args: Vec<OsString>) -> anyhow::Result<(Self, Vec<OsString>)> {
+        use lexopt::prelude::*;
+
+        let mut parser = lexopt::Parser::from_args(args);
+        let mut urn: Option<Urn> = None;
+
+        while let Some(arg) = parser.next()? {
+            match arg {
+                Long("help") => {
+                    return Err(Error::Help.into());
+                }
+                Value(val) if urn.is_none() => {
+                    let val = val.to_string_lossy();
+
+                    if let Ok(val) = Urn::from_str(&val) {
+                        urn = Some(val);
+                    } else {
+                        return Err(anyhow!("invalid URN '{}'", val));
+                    }
+                }
+                _ => return Err(anyhow::anyhow!(arg.unexpected())),
+            }
+        }
+
+        Ok((Options { urn }, vec![]))
+    }
+}
+
+pub fn run(options: Options) -> anyhow::Result<()> {
+    let profile = profile::default()?;
+    let signer = term::signer(&profile)?;
+    let storage = keys::storage(&profile, signer)?;
+
+    let urn = options
+        .urn
+        .or_else(|| radicle_common::project::cwd().ok().map(|(urn, _)| urn))
+        .ok_or_else(|| anyhow!("Couldn't get URN from either command line or cwd"))?;
+
+    let identity = any::get(&storage, &urn)?
+        .ok_or_else(|| anyhow!("No project or person found for this URN"))?;
+
+    let payload = match identity {
+        SomeIdentity::Project(_) => {
+            serde_json::to_string_pretty(project::verify(&storage, &urn)?.unwrap().payload())?
+        }
+        SomeIdentity::Person(_) => {
+            serde_json::to_string_pretty(person::verify(&storage, &urn)?.unwrap().payload())?
+        }
+        _ => todo!(),
```

You may want to throw an error message here. Hitting this case would mean that radicle-link has added a new identity type.

---

`anyhow!("unknown identity type for {}", urn)`